app/pages/sections/additional-services/order-item/flat/patient/ui-tests/withSavedData.ui.test.js

```diff
@@ -0,0 +1,238 @@
+import nock from 'nock';
+import { ClientFunction, Selector } from 'testcafe';
+import { extractInnerText } from 'buying-catalogue-library';
+import { orderApiUrl } from '../../../../../../../config';
+import content from '../manifest.json';
+
+const pageUrl = 'http://localhost:1234/order/organisation/order-1/additional-services/item-1';
+
+const getLocation = ClientFunction(() => document.location.href);
+
+const orderItem = {
+  serviceRecipient: {
+    odsCode: 'OX3',
+    name: 'Some service recipient 2',
+  },
+  catalogueItemType: 'Additional Service',
+  catalogueItemName: 'Some item name',
+  catalogueItemId: '10000-001',
+  quantity: 3,
+  estimationPeriod: 'month',
+  provisioningType: 'Patient',
+  type: 'flat',
+  currencyCode: 'GBP',
+  itemUnit: {
+    name: 'consultation',
+    description: 'per consultation',
+  },
+  price: 0.1,
+};
+
+const authTokenInSession = JSON.stringify({
+  id: '88421113', name: 'Cool Dude', ordering: 'manage', primaryOrganisationId: 'org-id',
+});
+const orderItemPageDataInSession = JSON.stringify({
+  itemId: orderItem.catalogueItemId,
+  itemName: orderItem.catalogueItemName,
+  serviceRecipientId: orderItem.serviceRecipient.odsCode,
+  serviceRecipientName: orderItem.serviceRecipient.name,
+  selectedPrice: {
+    price: orderItem.price,
+    itemUnit: orderItem.itemUnit,
+    type: orderItem.type,
+    provisioningType: orderItem.provisioningType,
+  },
+});
+
+const setState = ClientFunction((key, value) => {
```

Comment:
I think this is a better way of doing it. Could this be a common function?

Reply:
I'll give that a go. I think it can but sometimes testCafe is bit difficult with doing stuff like this.

Reply:
unfortunately it's not that straight forward. We need to pass in the ClientFunction also from the ui.test which makes it bit tricky to do 😞 .

Will leave this as is for now.